### ISSUE TITLE
fields.image missing

### ISSUE THREAD

**Issue by WillSquire** [keystonejs/keystone#7128]
I just dived into your framework and was so impressed by it that I wanted to migrate my site immediately. However, I've hit a bit of a snag. The component blocks don't appear to allow for images, but my hero components need them to set a background. I'm expecting it should follow suit with your fields API?

If anyone else has the same issue, I'm thinking of creating an image entity on the schema and creating a relationship instead 🤞.

**Comment by dcousens** [comment]
@WillSquire do I understand that your question is

> How do I add images as (or to?) component blocks?

**Comment by WillSquire** [comment]
@dcousens Yes. I'd expect images to be able to be uploaded in the component blocks with something like `fields.image({ label: '...' })`. I managed to find a hero example in your docs, but it links to an image. I'd expect an image field https://keystonejs.com/docs/guides/document-fields#fields so users can upload a hero image (or whatever component might require it). Also couldn't find a way to handle an array or repeat content - such as a list. It's a great framework, but it has been blocked for me.

**Comment by WillSquire** [comment]
No, but is there an example of this? If I can create a Hero entity inline (without going off to make the entity first, then referencing it), it could work, but the fear is there would be many 'one time' entities produced just for one use case or one 'page'. 

Because many entities will only be produced once, it feels like a `fields.list` (as well as `fields.image`) should exist, as `fields.object` does for component blocks. I generally don't want a 'Heros' entity listed on the main content list, as it should just reside as a component block that's used once for a specific page.

**Comment by LouisTrezzini** [comment]
Hello! 

Completely agree with @WillSquire. We love Keystone, but the lack of ability to upload images in the rich text document (both inline and inside component blocks) is a deal breaker for us. Asking content editors to upload the images then paste a link into an "image" component block is just too impractical.
 
Is there a workaround, or a plan to add this capability soon?

Thanks!

**Comment by gautamsi** [comment]
you want to take a look at this - https://github.com/jutiss/keystone6-document-gallery-block
ref https://keystonejs.slack.com/archives/C01STDMEW3S/p1641617614089500?thread_ts=1641598526.088100&cid=C01STDMEW3S

**Comment by LouisTrezzini** [comment]
Interesting, thanks!

While this looks great, this doesn't really work for us:
- This requires to create an "Image" list. While this is fine, it's inconsistent with the treatment of other images in Keystone, and I would expect content editors to be confused. If we want an "Image" list, I'd like to see all images in keystone belong to this list.
- This only somewhat solves the issue of not being able to add inline images to documents. However, if we want to build rich component blocks that have image fields, we can't use the Gallery component and still need to rewrite a huge amount of code (eg image with caption, user quote with attribution and user picture, etc)

**Comment by gautamsi** [comment]
well that gives you starting point on how to create your own component and share with open source world if you like
